File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
#   배열의 크기가 백만? 그럼 O(N)정도로 짜야할 듯..
#  answer 리스트에 추가하고, 같으면 continue하고 다르면 추가해서 리스트 1바퀴돌면 되지 않을까?


    answer = []
    answer.append(arr[0])
    k = 0
    for i in range(len(arr)):
        if answer[k] != arr[i]:
            k+=1
            answer.append(arr[i])
    return answer

#     for i in range(len(arr)):
#         if arr[i] not in   :
#             answer.append(arr[i])
#         else :
#             if i > 0 :
#                 if arr[i-1] != arr[i]:
#                     answer.append(arr[i])
#             continue
    
#     return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    /*
    arr = [1,1,3,3,0,1,1]이면
    분류 : 스택
    어떻게 풀이?
    1. 맨 처음 배열의 숫자를 스택에 넣는다.
    2. 그 이후부터는 스택의 맨 위 숫자와 같으면 무시하고, 다르면 넣는다
    
    */
    public int[] solution(int []arr) {
        Deque<Integer> st = new ArrayDeque<>();
        
        for (int x : arr) {
            if (st.isEmpty() || st.peekLast() != x) {
                st.addLast(x);
            }
        }
        
        int[] answer = st.stream()
              .mapToInt(x->x)
              .toArray();

        return answer;
    }
}